app/controllers/merchants/invoices_controller.rb

```diff
@@ -1,5 +1,23 @@
-class InvoicesController < ApplicationController
+class Merchants::InvoicesController < ApplicationController
 
   def index
+    @merchant = Merchant.find(params[:merchant_id])
+    @invoices = @merchant.invoices.uniq
+  end
+
+  def show
+    @merchant = Merchant.find(params[:merchant_id])
+    @invoice = Invoice.find(params[:id])
+    @invoice_items = @invoice.invoice_items
```

Comment:
Was this the data that kept rendering in a different order when rendering a new view?